fvtest/compilertriltest/IficmpTest.cpp

```diff
@@ -0,0 +1,174 @@
+/*******************************************************************************
+ *
+ * (c) Copyright IBM Corp. 2017, 2017
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ ******************************************************************************/
+
+#include <gtest/gtest.h>
+#include "Jit.hpp"
+#include "jitbuilder_compiler.hpp"
+
+#define ASSERT_NULL(pointer) ASSERT_EQ(nullptr, (pointer))
+#define ASSERT_NOTNULL(pointer) ASSERT_TRUE(nullptr != (pointer))
+
+extern "C" bool initializeJitWithOptions(char *options);
+
+using IfcmpReductionParamType = std::tuple<const char*, const char*, const char*, const char*>;
+
+class IfcmpReductionTest : public ::testing::TestWithParam<IfcmpReductionParamType>
+   {
+   public:
+
+   static void SetUpTestCase()
+      {
+      std::string defaultOptions = "-Xjit:acceptHugeMethods,enableBasicBlockHoisting,omitFramePointer,useIlValidator,paranoidoptcheck";
+      // Disable the following optimizations; otherwise, the trees are optimized into nothing.
+      std::string userOptions = "disableGlobalVP,disableLocalCSE";
+      std::string options = defaultOptions + "," + userOptions;
+
+      auto initSuccess = initializeJitWithOptions((char*)options.c_str());
+      ASSERT_TRUE(initSuccess) << "Failed to initialize the JIT.";
+      }
+
+   static void TearDownTestCase()
+      {
+      shutdownJit();
+      }
+   };
+
+/*
+ * Test reductions of ifcmpge/le opcodes to ifmcpeq
+ *
+ * Currently, we cannot perform this test automatically, so we are resorted to
+ * manually running the tests and examining the trace file.
+ *
+ * Ideally, the following should be performed:
+ * 1) Construct the trees that are to be transformed
+ * 2) Do the transformation
+ * 3) Compare the resulting trees with the expected trees.
+ *
+ * However, the current implementation of Tril does not support step 3).
+ *
+ * To produce a trace log, run the following:
+ * $ TR_Options=traceilgen,tracefull,log=trace.log ./comptest --gtest_filter="IfcmpReduction/IfcmpReductionTest.Reduction/ByteSignedLeReducible"
+ *
+ * Then, open the trace log and witness the transformation.
+ */
+TEST_P(IfcmpReductionTest, Reduction)
+   {
+   auto param = GetParam();
+
+   const char *type = std::get<0>(param);
+   const char *sign = std::get<1>(param);
+   const char *lege = std::get<2>(param);
+   const char *val  = std::get<3>(param);
```

Comment:
Since you access the tuple both here and in `reductionTestNameGenerator()`, have you considered defining a struct that gives names to the tuple elements and defining a `to_struct()` function that converts an `IfcmpReductionParamType` to a struct instance (similar to `BinaryOpParamType` and `BinaryOpParamStruct` in `OpCodeTest.hpp`)?